### PR TITLE
fix: Fix and test null values Pentabarf-xml.py

### LINKED ISSUE #6513: Pentabarf export not working
https://github.com/fossasia/open-event-server/blob/97e94f2ce14d20341253822b5456c93bef590c71/app/api/helpers/pentabarfxml.py#L61-L64

Line 64: `unsupported operand type(s) for -: 'NoneType' and 'NoneType'`

### PR DESCRIPTION
**Pull request by PalAditya**
Fixes #6513 

#### Short description of what this resolves:
This prevents crashing of the export when either start time or end time of the event is null

#### Changes proposed in this pull request:

- Check whether either start time or end time is null
- If so, set the duration as "Not set", else let it continue as before
- Note: I have assumed that if the values are not null, they will be capable of being subtracted. A more rigorous check can be done by using the isinstance method on the two parameters and ensuring they are int/floats, but I was not sure it was necessary, and if so, please mention it.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Please add a test for this as well and change the PR title according to the guidelines

**Comment by PalAditya** [comment]
@iamareebjamal , When I actually try to insert a Null value, I get caught by the integrity check (As I mentioned, this is probably expected due to the constraints)
A basic travis log:
``` 
cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.errors.NotNullViolation) null value in column "starts_at" violates not-null constraint
```

Hence, I am at a loss as to how to test that method as the constraint prevents me from nulls, and I wonder how NoneType values got into that column in the first place (is it possible that the constraint was added later?)
Any advice? 😅

**Comment by mrsaicharan1** [comment]
> @iamareebjamal , When I actually try to insert a Null value, I get caught by the integrity check (As I mentioned, this is probably expected due to the constraints)
> A basic travis log:
> 
> ```
> cursor.execute(statement, parameters)
> sqlalchemy.exc.IntegrityError: (psycopg2.errors.NotNullViolation) null value in column "starts_at" violates not-null constraint
> ```
> 
> Hence, I am at a loss as to how to test that method as the constraint prevents me from nulls, and I wonder how NoneType values got into that column in the first place (is it possible that the constraint was added later?)
> Any advice? 😅

Reset the DB and start afresh.

**Comment by PalAditya** [comment]
@mrsaicharan1 , I don't think resetting the db will do anything: after all, what I need is to insert is None at some columns, which are prevented by the Integrity check (now), but I was told that those columns actually contained null values, so I am waiting for an explanation as to how that could have happened 🤔

**Comment by mrsaicharan1** [comment]
Are you talking about the prod DB values or values on your local DB?

**Comment by mrsaicharan1** [comment]
@paladitya
There might’ve been schema level changes after the values were added into the DB. Try to find out the reason. Maybe diving deep into the commit history could help.

**Comment by PalAditya** [comment]
> Are you talking about the prod DB values or values on your local DB?

Talking about the prod db, since I was told that contains null values. However, my local db doesn't allow me to enter null values, which is the problem as I can't write tests for it in that case.
I will look into the commit history and try to figure it out. Thanks for the advice!

File: tests/all/integration/api/helpers/test_pentabarfxml.py [hunk 1]
```diff
@@ -23,6 +23,20 @@ def test_export(self):
...
+        """Test to check event contents in pentabarfxml format"""
+        with app.test_request_context():
+            test_event = EventFactoryBasic()
+            test_event.starts_at = None
+            test_event.ends_at = None
+            save_to_db(test_event)
+            pentabarf_export = PentabarfExporter()
+            pentabarf_string = pentabarf_export.export(test_event.id)
+            pentabarf_original = fromstring(pentabarf_string)
+            print(pentabarf_original)
```
Comment (iamareebjamal) [inline tests/all/integration/api/helpers/test_pentabarfxml.py]: We can remove this statement here

**Comment by iamareebjamal** [comment]
Yeah, the schema defined in the app and actual production DB are not in sync unfortunately

**Comment by iamareebjamal** [comment]
You should remove nullable False from the starts_at and ends_at, and create a new migration

File: tests/all/integration/api/helpers/test_pentabarfxml.py (hunk 1 above)
Comment (PalAditya) [reply tests/all/integration/api/helpers/test_pentabarfxml.py]: Yes, sorry about the ```print```

**Comment by PalAditya** [comment]
> You should remove nullable False from the starts_at and ends_at, and create a new migration

Do you mean that I should make those changes for this PR itself (which will then allow null values to be passed) and then ensure that the tests pass? I am not sure about the term ```migration```.

Will do that then 😄

**Comment by mrsaicharan1** [comment]
Read the repo docs on how DB migrations can be done

**Comment by iamareebjamal** [comment]
@PalAditya Updates? We need to release a version soon

**Comment by PalAditya** [comment]
> @PalAditya Updates? We need to release a version soon

No, sorry. I have been very busy this past week due to moving to a new place and am still settling down, so please feel free to assign the issue to somebody else and close this PR in case the release is urgent.

I am truly sorry about this 😅

**Comment by iamareebjamal** [comment]
@mrsaicharan1 Can you complete this?

**Comment by iamareebjamal** [comment]
Superseded by #6533

**Pull request closed without merging**